cypress/integration/admin/pageBuilder/blocks/pageBlocks.cy.ts

```diff
@@ -0,0 +1,57 @@
+import { nanoid } from "nanoid";
+import { customAlphabet } from "nanoid";
+
+context("Blocks Page", () => {
+    let tokenStorage;
+    const nanoid = customAlphabet("abcdefghijklmnopqrstuvwxyz");
+    const blockCategoryName = nanoid(10); // Generate a random 10-character lowercase string
+    const slugName = nanoid(10); // Generate another random 10-character lowercase string
+
+    beforeEach(() => cy.login());
+    beforeEach(() => cy.pbDeleteBlocks());
+    beforeEach(() => cy.pbAllDeleteBlockCategories());
+
+    it("Should be able to create a block category and a block and then edit, duplicate and delete it", () => {
+        cy.visit("/page-builder/page-blocks");
+        cy.findByPlaceholderText("Search blocks").should("exist");
+        cy.findByTestId("pb-blocks-list-new-block-btn").click();
+        cy.findByTestId("pb-blocks-list-new-block-category-btn").click();
+
+        cy.findByRole("textbox", { name: "Name" }).type(blockCategoryName);
+        cy.findByRole("textbox", { name: "Slug" }).type(slugName);
+        cy.findByRole("textbox", { name: "Description" }).type(blockCategoryName);
+        cy.findByTestId("pb-block-categories-form-save-block-category-btn").click();
+
+        cy.visit("/page-builder/page-blocks");
+        cy.findByPlaceholderText("Search blocks").should("exist");
+        cy.findByTestId("pb-blocks-list-new-block-btn").click();
+        cy.findByText(slugName).click();
+        cy.findByTestId("pb-blocks-editor-save-changes-btn").click();
+        cy.findByTestId("pb-blocks-list-new-block-btn").should("exist");
```

Comment:
BTW let's also have the edit button tested. Once a block has been created, let's ensure that when we click on the edit button, the user is redirected to the editor, and that in the top-left corner, the correct block name is shown.